internal/logstore/lokistack/logstore_lokistack.go

```diff
@@ -242,12 +242,16 @@ func getInputTypeFromName(spec loggingv1.ClusterLogForwarderSpec, inputName stri
 		return input.Receiver != nil && input.Receiver.HTTP != nil && input.Receiver.HTTP.Format == format
 	}
 
+	isSyslogReceiver := func(input loggingv1.InputSpec) bool {
+		return input.Receiver != nil && input.Receiver.Syslog != nil
+	}
+
 	for _, input := range spec.Inputs {
 		if input.Name == inputName {
 			if input.Application != nil {
 				return loggingv1.InputNameApplication
 			}
-			if input.Infrastructure != nil {
+			if input.Infrastructure != nil || isSyslogReceiver(input) {
```

Comment:
This smells wrong to me and outside the design of LogForwarding as we know it.  @alanconway there has to be a better way to identify the tenant.  Maybe This is where we need to allow tenant configuration..... and i see we are doing that below as well for http which smells wrong

Reply:
Indeed, I am here repeating the pattern established for HTTP Receiver, as @alanconway told me.

Reply:
Agreed, this does suck. We should configure log_type consistently, possibly this is one and only top-level configuration setting for ReceiverSpec? We have no other way to assign a log_type for receiver logs. We need to discuss this, I see two options:
1. Let user assign log_type, allow them to "impersonate" built-in types and/or allow them to invent new types and filter them for themselves.
2. Invent a fouth fixed type like "external" and stamp that on received logs. Let the user use other features (like attaching openshift.labels) to identify where the logs came from.
I'm leaning towarsd 1., since I think we want to de-emphasize log_type and focus on filtering/routing by other log properties, but I can also see arguments for 2.

Reply:
As per our latest meeting, we are leaving the "isSyslogReceiver" here to be able to provide logging storage for Openstack until we deliver a better solution for version 6 of CLO.